pkg/sriov/sriov.go

```diff
@@ -128,13 +135,22 @@ func (s *sriovManager) SetupVF(conf *sriovtypes.NetConf, podifName string, cid s
 		return "", fmt.Errorf("failed to move IF %s to netns: %q", tempName, err)
 	}
 
-	// 4. Set Pod IF name
 	if err := netns.Do(func(_ ns.NetNS) error {
+		// 4. Set Pod IF name
 		if err := s.nLink.LinkSetName(linkObj, podifName); err != nil {
 			return fmt.Errorf("error setting container interface name %s for %s", linkName, tempName)
 		}
 
-		// 5. Bring IF up in Pod netns
+		// 5. Restore pod's original IP if IPAM is missing
+		if conf.IPAM.Type != "" && len(addrs) > 0 {
```

Comment:
@rajatchopra 

Should it be conf.IPAM.Type == "" ?

no Need for len(addrs), range in for loop will take care of it for empty addrs.

Reply:
Sorry about that. I tested one set and pushed another set of code :).